src/averageAges.js

```diff
@@ -1,62 +1,42 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function getAge(deathYear, birthYear) {
+  return deathYear - birthYear;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person =>
+    person.sex === 'm' && (century
+      ? Math.ceil(person.died / 100) === century
+      : true));
+
+  return men.reduce((acc, man) =>
+    getAge(man.died, man.born) + acc, 0) / men.length;
```

Comment:
Не розривай колбеки на інші рядки, так важко зрозуміти, де функція закінчується
```suggestion
  return men.reduce(
  (acc, man) => getAge(man.died, man.born) + acc
  , 0) / men.length;
```

Reply:
fixed